internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer.go

```diff
@@ -40,6 +40,31 @@ func (p PhaseList) AppendWhen(when bool, name string, tasks ...runtime.TaskSetup
 	return p
 }
 
+// ApplyConfiguration is a sequence which is called to apply a new configuration to the system.
+// It will reboot subsequent make it active.
+func (*Sequencer) ApplyConfiguration(r runtime.Runtime, req *machineapi.ApplyConfigurationRequest) []runtime.Phase {
+	phases := PhaseList{}
+
+	phases = phases.Append(
+		"mountState",
+		MountStatePartition,
+	).Append(
+		"saveConfig",
+		SaveConfig,
+	).Append(
+		"unmountState",
+		UnmountStatePartition,
+	).Append(
+		"stopEverything",
+		StopAllServices,
+	).Append(
```

Comment:
It could be argued that we should not bind a reboot implicitly with the application of a new configuration.  I think we should do so, at least by default, but I could see opposing arguments.

Reply:
I like the idea of rebooting on config apply.

Only one concern is what happens if config is bad, so node might not boot back, and in that case it seems as dangerous as upgrade. I don't want to put it on this PR, but just a thought.